service/worker/archiver/activities.go

```diff
@@ -62,8 +63,9 @@ func uploadHistoryActivity(ctx context.Context, request ArchiveRequest) (err err
 		if err != nil {
 			if err.Error() == errUploadNonRetryable.Error() {
 				scope.IncCounter(metrics.ArchiverNonRetryableErrorCount)
+			} else {
+				err = temporal.NewApplicationError(err.Error(), "", nil)
```

Comment:
This is default `error` to `ApplicationError` conversion. I would just return `err` here.
```suggestion
```

Reply:
And everywhere bellow.

Reply:
Makes sense. Then no change is needed to `err` and now the defer looks like the following:

```
	defer func() {
		sw.Stop()
		if err == errUploadNonRetryable {
			scope.IncCounter(metrics.ArchiverNonRetryableErrorCount)
		}
	}()
```